java/java.hints/src/org/netbeans/modules/java/hints/jdk/Regexp.java

```diff
@@ -0,0 +1,185 @@
+    public static ErrorDescription hint(final HintContext ctx) {
+        final StringBuilder regexp = new StringBuilder();
+        final boolean[] accept = {true};
+        Tree leaf = ctx.getVariables().get("$pattern").getLeaf();
+        if (leaf.getKind() == Kind.STRING_LITERAL) {
+            TreePath pattern = ctx.getVariables().get("$pattern");
+            new ErrorAwareTreePathScanner<Void, Void>() {
+                @Override
+                public Void visitLiteral(LiteralTree node, Void p) {
+                    if (node.getValue() instanceof String) {
+                        regexp.append(node.getValue());
+                        return null;
+                    }
+                    accept[0] = false;
+                    return null;
+                }
+
+                @Override
+                public Void visitIdentifier(IdentifierTree node, Void p) {
+                    Element el = ctx.getInfo().getTrees().getElement(getCurrentPath());
+
+                    if (el != null && el.getKind() == ElementKind.FIELD) {
+                        VariableElement ve = (VariableElement) el;
+
+                        if (ve.getConstantValue() instanceof String) {
+                            regexp.append(ve.getConstantValue());
+                            return null;
+                        }
+                    }
+                    accept[0] = false;
+                    return null;
+                }
+
+                @Override
+                public Void visitMemberSelect(MemberSelectTree node, Void p) {
+                    Element el = ctx.getInfo().getTrees().getElement(getCurrentPath());
+
+                    if (el != null && el.getKind() == ElementKind.FIELD) {
+                        VariableElement ve = (VariableElement) el;
+
+                        if (ve.getConstantValue() instanceof String) {
+                            regexp.append(ve.getConstantValue());
+                            return null;
+                        }
+                    }
+                    accept[0] = false;
+                    return null;
+                }
+
+                @Override
+                public Void visitBinary(BinaryTree node, Void p) {
+                    if (node.getKind() != Kind.PLUS) {
+                        return super.visitBinary(node, p);
+                    }
+                    accept[0] = false;
+                    return null;
+                }
+
+                @Override
+                public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
+                    accept[0] = false;
+                    return null;
+                }
+            }.scan(pattern, null);
+
+            if (!accept[0] || regexp.length() == 0) {
+                return null;
+            }
+        }else if(leaf.getKind() == Kind.IDENTIFIER){
+            String val = CheckRegex.identifierSearch(leaf, ctx);
+            if(val != null)
+                regexp.append(val);
+        }
```

Comment:
Please reformat.

Reply:
Thanks. Have made the changes and checked the formatting of the files.